src/common/com/intellij/plugins/haxe/lang/psi/impl/HaxePsiFieldImpl.java

```diff
@@ -62,25 +64,16 @@ public String getName() {
   }
 
   @Override
-  @NotNull
+  @Nullable
   public HaxeComponentName getComponentName() {
-    return new HaxeComponentNameImpl(getNode());
-    ////this.getName();
-    //return findNotNullChildByClass(HaxeComponentName.class);
+    return null;
```

Comment:
getNode() was right, and NameComponentImpl was created, and then NameComponentImpl::getIdentifier - `findNotNullChildByClass(HaxeIdentifier.class)` throws exception because of `findNotNull__` guarding. I don't remember the case exactly. I could spend a little amount of time to reproduce this case.

NPE had beed shown up for VarDeclaration, not VarPartDeclaration. Notice that getComponentName is overridden for all Implementations, except of VarDeclaration(part is overridden). So we've guarded `getNameIdentifier` which will be Null for VarDeclaration (because the component name inside VarPart).